nuto/mechanics/solver/Solve.cpp

```diff
@@ -1,23 +1,99 @@
 #include "Solve.h"
 #include "nuto/math/EigenSparseSolve.h"
+#include "nuto/mechanics/dofs/DofVectorConvertEigen.h"
+#include "nuto/mechanics/dofs/DofMatrixSparseConvertEigen.h"
 
-namespace NuTo
+using namespace NuTo;
+
+DofVector<double> NuTo::Solve(const DofMatrixSparse<double>& K, const DofVector<double>& f,
+                              Constraint::Constraints& bcs, std::vector<DofType> dofs, std::string solver)
 {
+    auto K_full = ToEigen(K, dofs);
+    auto f_full = ToEigen(f, dofs);
+
+    DofMatrixSparse<double> C_dof;
+    for (auto rdof : dofs)
+        C_dof(rdof, rdof) = bcs.BuildUnitConstraintMatrix2(rdof, f[rdof].rows());
+
+    for (auto rdof : dofs)
+        for (auto cdof : dofs)
+            if (rdof.Id() != cdof.Id())
+                C_dof(rdof, cdof) = Eigen::SparseMatrix<double>(C_dof(rdof, rdof).rows(), C_dof(cdof, cdof).cols());
 
-GlobalDofVector Solve(GlobalDofMatrixSparse K, GlobalDofVector f, Constraint::Constraints bcs, DofType dof,
-                      int numIndependentDofs, double time, std::string solver)
+    auto C = ToEigen(C_dof, dofs);
+    Eigen::SparseMatrix<double> Kmod = C.transpose() * K_full * C;
```

Comment:
I agree that this conversion to and from eigen is very messy. However, I would not prefer hiding this conversion in another layer (in this operator there is either a similar conversion or a manual implementation of the multiplication).

Reply:
I suggest to avoid the conversion and perform the _manual implementation_. But I see no problem in that. We also did a manual implementation of `operator+` and `operator-`.